### ISSUE TITLE
Attention! There is a vulnerability in this library that could be exploited. I urge the author to fix it as soon as possible!

### ISSUE THREAD

**Issue by jarryleo** [hannesa2/paho.mqtt.android#492]
Hello, I have found the cause of the bug from yesterday. It is possible that another app is also using your library but with a different version of MQTT, which leads to the crash. It could also be a malicious attack.
In your library, there is a broadcast that is not isolated to the app, so external apps can also receive it. If an external app sends a broadcast with a custom MqttException, all apps using your library will crash.
I have created a demo app and sent the following broadcast, and as a result, all apps using this library crashed:
I suggest that you avoid using global broadcasts in your library and instead use local broadcasts.
```
//custom MqttException
class MqttException(message: String?) : Exception(message) {
    companion object {
        private const val serialVersionUID = 20230804L
    }
}

fun attack(context: Context) {
    val callbackIntent = Intent(".callbackToActivity.v0")
    val bundle = Bundle()
    bundle.putSerializable(".exception", MqttException("mqttAttack"))
    callbackIntent.putExtras(bundle)
    context.sendBroadcast(callbackIntent)
}
```

_Originally posted by @jarryleo in https://github.com/hannesa2/paho.mqtt.android/issues/490#issuecomment-1665104518_

**Comment by hannesa2** [comment]
Thank you to pint this out. 
Local broadcasts are deprecated https://developer.android.com/jetpack/androidx/releases/localbroadcastmanager#1.1.0-alpha01 and I already tried to work on this #401 and #391 but with no success till now.

If you want a faster solution, you can help with this too

**Comment by hannesa2** [comment]
closed with #495
